File: 49-group-anagrams/group-anagrams.py

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        Map=defaultdict(list)
        for word in strs:
            freq=[0]*26
            for ch in word:
                freq[ord(ch)-ord('a')]+=1

            key=tuple(freq)    
            Map[key].append(word)

        return list(Map.values())            

File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<int>>Map;
        for(int i=0;i<strs.size();i++){
            string str=strs[i];
            sort(str.begin(),str.end());
            Map[str].push_back(i);
        }
        vector<vector<string>>ans;
        for(auto st : Map){
            vector<string>str;
            for(auto i : st.second){
                str.push_back(strs[i]);
            }
            ans.push_back(str);
        }
        return ans;
        
    }
};